pkg/controller/workload/workload_processor.go

```diff
@@ -486,6 +478,26 @@ func (p *Processor) storeServiceData(serviceName string) error {
 	sk.ServiceId = p.hashName.StrToNum(serviceName)
 	sv.LbPolicy = LbPolicyRandom
 	sv.EndpointCount = 0 // there are 0 endpoints in the initial state
+
+	if waypoint != nil {
+		sv.WaypointAddr = nets.ConvertIpByteToUint32(waypoint.GetAddress().Address)
+		sv.WaypointPort = nets.ConvertPortToBigEndian(waypoint.GetHboneMtlsPort())
+	}
+
+	for i, port := range ports {
+		if i >= bpf.MaxPortNum {
+			log.Warnf("exceed the max port count,current only support maximum of 10 ports")
+			break
+		}
+
+		sv.ServicePort[i] = nets.ConvertPortToBigEndian(port.ServicePort)
+		if strings.Contains(serviceName, "waypoint") {
```

Comment:
Maybe incorrect judgment if ServiceName contains `waypoint`, but actually it is not `waypoint` Pod

Reply:
yeah, there is indeed such a problem at present, and we will track the relevant optimization situation through this [issue](https://github.com/kmesh-net/kmesh/issues/370) in the future.